services/order.py

```diff
@@ -0,0 +1,31 @@
+from django.contrib.auth import get_user_model
+from django.db.models import QuerySet
+from django.db import transaction
+from db.models import Ticket, Order
+
+
+@transaction.atomic
+def create_order(tickets: list[dict],
+                 username: str,
+                 date: str = None
+                 ) -> QuerySet:
+    user = get_user_model().objects.get(username=username)
+    order = Order.objects.create(user=user)
+
+    if date:
+        order.created_at = date
+        order.save()

```

Comment:
You're calling the `save()` method twice in the `create_order` function. It would be better to call it only once at the end of the function. This is a violation of the 'Don't Repeat Yourself' principle from the checklist.